photon-lib/src/main/java/org/photonvision/RobotPoseEstimator.java

```diff
@@ -257,7 +257,16 @@ private Pair<Pose3d, Double> averageBestTargetsStrategy() {
                     continue;
                 }
                 Pose3d targetPose = aprilTags.get(target.getFiducialId());
-                totalAmbiguity += target.getPoseAmbiguity();
+                try {
+                    totalAmbiguity += 1. / target.getPoseAmbiguity();
+                } catch (ArithmeticException e) {
+                    DriverStation.reportWarning(
+                            "[RobotPoseEstimator] A total ambiguity of zero exists, using that pose instead!",
+                            false);
+                    return Pair.of(
+                            targetPose.transformBy(target.getBestCameraToTarget().inverse()),
+                            p.getFirst().getLatestResult().getLatencyMillis());
+                }
```

Comment:
this try catch is slightly curst lol

Reply:
Yeah; we want to do an inversed weighted average (lower ambiguity = higher weight), so the totalAmbiguity needs to be the summation of the inverse ambiguities; reduces time complexity in the 0 edge case 😉  .